metrics/resource_view.go

```diff
@@ -51,14 +54,51 @@ type meters struct {
 	meters  map[string]*meterExporter
 	factory ResourceExporterFactory
 	lock    sync.Mutex
+	clock   clock.Clock
+	ticker  clock.Ticker
 }
 
 // Lock regime: lock allMeters before resourceViews. The critical path is in
 // optionForResource, which must lock allMeters, but only needs to lock
 // resourceViews if a new meter needs to be created.
-var resourceViews = storedViews{}
-var allMeters = meters{
-	meters: map[string]*meterExporter{"": &defaultMeter},
+var (
+	resourceViews = storedViews{}
+	allMeters     = meters{
+		meters: map[string]*meterExporter{"": &defaultMeter},
+		clock:  cleanupClock,
+	}
+
+	cleanupClock               = clock.Clock(clock.RealClock{}) // Var needed to avoid allMeters->defaultMeter->allMeters.clock cycle
+	cleanupOnce                = new(sync.Once)
+	meterExporterScanFrequency = 1 * time.Minute
+	maxMeterExporterAge        = 10 * time.Minute
+)
+
+func cleanup() {
+	log.Printf("in cleanup")
+	allMeters.lock.Lock()
+	defer allMeters.lock.Unlock()
+	expiryCutoff := allMeters.clock.Now().Add(-1 * maxMeterExporterAge)
+	for key, meter := range allMeters.meters {
+		if key != "" && meter.t.Before(expiryCutoff) {
```

Comment:
This is probably worth a comment about how it's avoiding a memory leak from Resources that are no longer monitored / emitting statistics.

Reply:
Done.